addons/mrp_byproduct/models/mrp_production.py

```diff
@@ -0,0 +1,53 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+from openerp import api, models
+
+
+class MrpProduction(models.Model):
+    _description = 'Production'
+    _inherit = 'mrp.production'
+
+    @api.multi
+    def action_confirm(self):
+        """ Confirms production order and calculates quantity based on subproduct_type.
+        """
+        StockMove = self.env['stock.move']
+        for production in self:
+            source = production.product_id.property_stock_production.id
+            if not production.bom_id:
+                continue
+            for sub_product in production.bom_id.sub_products_ids:
+                product_uom_factor = production.product_uom._compute_qty(production.product_qty, production.bom_id.product_uom.id)
+                qty1 = sub_product.product_qty
+                if sub_product.subproduct_type == 'variable':
+                    if production.product_qty:
+                        qty1 *= product_uom_factor / (production.bom_id.product_qty or 1.0)
+                data = {
+                    'name': 'PROD:' + production.name,
+                    'date': production.date_planned,
+                    'product_id': sub_product.product_id.id,
+                    'product_uom_qty': qty1,
+                    'product_uom': sub_product.product_uom_id.id,
+                    'location_id': source,
+                    'location_dest_id': production.location_dest_id.id,
+                    'move_dest_id': production.move_prod_id.id,
+                    'production_id': production.id
+                }
+                StockMove.create(data).action_confirm()
+        return super(MrpProduction, self).action_confirm()
```

Comment:
are you sure to call super at last ? in old api it's called before we do this operation.